Source/WebCore/dom/Element.h

```diff
@@ -700,9 +699,12 @@ class Element : public ContainerNode {
     ResizeObserverData& ensureResizeObserverData();
     ResizeObserverData* resizeObserverData();
 
-    ResizeObserverSize* lastRememberedSize() const;
-    void setLastRememberedSize(Ref<ResizeObserverSize>&&);
-    void clearLastRememberedSize();
+    std::optional<LayoutUnit> lastRememberedLogicalWidth() const;
+    std::optional<LayoutUnit> lastRememberedLogicalHeight() const;
+    void setLastRememberedLogicalWidth(std::optional<LayoutUnit>);
+    void setLastRememberedLogicalHeight(std::optional<LayoutUnit>);
+    void clearLastRememberedLogicalWidth();
+    void clearLastRememberedLogicalHeight();
```

Comment:
I think its a bit strange to expose both `set*(std::optional)` and `clear*()`. I would either have

```cpp
void ElementRareData::setLastRememberedLogicalWidth(std::optional<LayoutUnit> width)
{
    m_lastRememberedLogicalWidth = width;
}

void Element::setLastRememberedLogicalWidth(LayoutUnit width)
{
    ensureElementRareData().setLastRememberedLogicalWidth(width);
}

void Element::clearLastRememberedLogicalWidth()
{
    if (auto* data = rareData())
        data->setLastRememberedLogicalWidth(std::nullopt);
}
```

or

```cpp
void ElementRareData::setLastRememberedLogicalWidth(LayoutUnit width)
{
    m_lastRememberedLogicalWidth = width;
}

void ElementRareData::clearLastRememberedLogicalWidth()
{
    m_lastRememberedLogicalWidth.reset();
}

void Element::setLastRememberedLogicalWidth(LayoutUnit width)
{
    ensureElementRareData().setLastRememberedLogicalWidth(width);
}

void Element::clearLastRememberedLogicalWidth()
{
    if (auto* data = rareData())
        data->clearLastRememberedLogicalWidth();
}
```

or

```cpp
void ElementRareData::setLastRememberedLogicalWidth(std::optional<LayoutUnit> width)
{
    m_lastRememberedLogicalWidth = width;
}

void Element::setLastRememberedLogicalWidth(std::optional<LayoutUnit> width)
{
    if (width)
        ensureElementRareData().setLastRememberedLogicalWidth(width);
    else if (auto* data = rareData())
        data->setLastRememberedLogicalWidth(std::nullopt);
}
```

Reply:
Done, I prefer to the last one:)

Reply:
Now that we use Markable<LayoutUnit>, I think we could reuse `clear*`, so that the outside don't have to use the magic number(-1).